testdata/testoutput2

```diff
@@ -20258,6 +20258,13 @@ No match
  2: 
  3: 
 
+/(a)(b+)(*scs:(1)a(*ACCEPT))(\2)/
+    abbb
+ 0: abb
```

Comment:
isn't this the current/expected output in 10.45 and HEAD?

Reply:
Apologies for not being clear enough, but the point I was trying to make is that the test case doesn't show the bug, at least on my system:

```
PCRE2 version 10.45 2025-02-05 (8-bit)
  re> /(a)(b+)(*scs:(1)a(*ACCEPT))(\2)/BI
------------------------------------------------------------------
        Bra
        CBra 1
        a
        Ket
        CBra 2
        b+
        Ket
        Scan substring
      1 Capture ref
        a
        *ASSERT_ACCEPT
        Ket
        CBra 3
        \2
        Ket
        Ket
        End
------------------------------------------------------------------
Capture group count = 3
Max back reference = 2
First code unit = 'a'
Subject length lower bound = 1
data> abbb
 0: abb
 1: a
 2: b
 3: b
```

Reply:
True. The `(PCRE2_SIZE)(mb->end_subject - eptr) < length` in line 488 is `18446744073709551613 < length` and does not true. A better test case is needed.

Reply:
the Mac install issue comes from an environmental change on the macOS runner that triggered a bug on our script that is addressed in the [first commit](https://github.com/PCRE2Project/pcre2/pull/777/commits/7496c0b835954845e4f87fe265ed4952ba55498f) of #777

Reply:
Normally that should never overflow. An assert could be added.

Reply:
OK, so the only old bug was the return value being incorrect (which might not that important anyway as it is unlikely to be user visible) and the fact that we didn't have an assert for eptr so this bug could go unnoticed?

Reply:
The old test case was better than the new one!

The test of `/(a)(b+)(*scs:(1)a(*ACCEPT))(\2)/` does reproduce the bug, but the updated test case does not.

The original test case crashes with both valgrind and clang address-sanitizer.

I will add both tests.

Reply:
Thank you for confirming! I will include only this fix in the 10.46 security release, in order to make minimal changes in this release.

Reply:
That's right. Changing the format of the version number to two digits has the potential to be disruptive to downstream consumers. Plus, we have API functions that expect a two-digit number, and the special VERSION pattern syntax...

It simply had to be 10.46.

Anyway, thanks to both of you for fixing this (Zoltan) and reviewing (Carlo).